content/posts/jenkins-x-flow-integration.md

```diff
@@ -0,0 +1,147 @@
+
+In the setup we’re going to choose the Kubernetes workloads option and later modify the kubernetes workload build packs to include the CloudBees Flow specific steps:
+
+{{< image src="/img/jenkins-x-flow-integration/jx-kubernetes-workload.jpg" alt="Jenkins X Kubernetes Workload" >}}
+
+## Editing the Build Packs
+I edited the Jenkins X build packs in VS Code and installed the YAML Language extension (https://jenkins-x.io/architecture/jenkins-x-pipelines/#editing-in-vs-code) for validation as recommended by the Jenkins X team.
+
+This example is going to focus on a sample spring boot application using Maven.  To start I modified the maven build pack in my forked build pack repo (https://github.com/jhendrickCB/jenkins-x-kubernetes/blob/master/packs/maven/pipeline.yaml):
+
+```bash
+extends:
+ import: classic
+ file: maven/pipeline.yaml
+pipelines:
+ release:
+   build:
+     steps:
+     - sh: jx step post build --image $DOCKER_REGISTRY/$ORG/$APP_NAME:\$(cat VERSION)
+       name: post-build
+   promote:
+     steps:
+     - sh: jx step changelog --version v\$(cat ../../VERSION)
+       name: changelog
+     - comment: call CloudBees Flow to run a release
+       sh: >
+         curl -X POST --header "Authorization: Basic $(jx step credential -s flow-token -k token)" --header "Content-Type: application/json" --header "Accept: application/json" -d "{}" "https://ps9.ecloud-kdemo.com/rest/v1.0/pipelines?pipelineName=jhendrick_pipeline&projectName=Training_jhendrick" --insecure
+       name: cloudbees-flow-release
+```
+
+Compare the original build pack for maven found here: https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes/tree/master/packs/maven vs. our forked build pack.  We’ve removed all the references to skaffold, watch, and helm since we’re no longer having Jenkin’s X handle the deployment to our Kubernetes cluster.  We’ve also updated the pipeline file to make a call into our CloudBees Flow server:
+
+```bash
+curl -X POST --header "Authorization: Basic $(jx step credential -s flow-token -k token)" --header "Content-Type: application/json" --header "Accept: application/json" -d "{}" "https://ps9.ecloud-kdemo.com/rest/v1.0/pipelines?pipelineName=jhendrick_pipeline&projectName=Training_jhendrick" --insecure
+```
+
+The above API call into CloudBees Flow tells Flow to run a pipeline called jhendrick_pipeline within a project called Training_jhendrick.
+
+You’ll also notice that we’re using a Jenkins X feature (jx step credential) to get our secret, flow-token, which we created previously so that we can authenticate to the Flow Rest API. Note that there are many other possible ways to call into the CloudBees Flow API’s besides cURL such as the command line tool ectool as well as python or groovy libraries.  Also note that for a production environment we would want to setup the proper certificates rather than using the --insecure parameter.
```

Comment:
Again jx step credential to `jx step credential`

flow-token `flow-token`
--insecure parameter `--insecure parameter`